client/sites-dashboard/use-sites-data-query.ts

```diff
@@ -0,0 +1,45 @@
+import config from '@automattic/calypso-config';
+import { useQuery } from 'react-query';
+import wpcom from 'calypso/lib/wp';
+import { SiteData, SiteDataOptions } from 'calypso/state/ui/selectors/site-data';
+
+// Performance-optimized request for lists of sites.
+// Don't add more fields because you will make the request slower.
+export const SITE_EXCERPT_REQUEST_FIELDS = [
+	'ID',
+	'URL',
+	'is_coming_soon',
+	'is_private',
+	'launch_status',
+	'slug',
+	'icon',
+	'name',
+	'options',
+	'plan',
+] as const;
+
+export const SITE_EXCERPT_REQUEST_OPTIONS = [ 'is_wpforteams_site' ] as const;
+
+export type SiteExcerptData = Pick< SiteData, typeof SITE_EXCERPT_REQUEST_FIELDS[ number ] > & {
```

Comment:
cool trick. Was gonna suggest this, we now change fields one place to add/remove fields for our excerpt.

Reply:
Yeah nice 😎 I haven't indexed into an array type using the `[ number ]` trick before 👍